x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -619,14 +620,14 @@ static Settings additionalSettings(final Settings settings, final boolean enable
     }
 
     @Override
-    public Collection<String> getRestHeaders() {
-        Set<String> headers = new HashSet<>();
-        headers.add(UsernamePasswordToken.BASIC_AUTH_HEADER);
+    public Collection<RestHeaderDefinition> getRestHeaders() {
+        Set<RestHeaderDefinition> headers = new HashSet<>();
+        headers.add(new RestHeaderDefinition(UsernamePasswordToken.BASIC_AUTH_HEADER, false));
```

Comment:
That’s an unrelated change that would then cause this PR to go and touch a bunch of files not related to the headers change, but instead the rename change. In general, we should avoid scope creep in a PR like that.

Reply:
To expand on this, it makes for confusing diffs, longer reviews, and it renders git blame less useful. Expanding on this last point: I go to git blame for a line impacted by this rename, and it would tell me it’s from the “Copy httpHeaders to ThreadContext strictly” which would leave me super confused because I wouldn’t have any context that it’s from a rename asked for during the review. I would then have to read the diff of that change to see it’s a rename that occurred in that changeset. If the rename is important, it should be a separate change, with a clear commit message. Then it would be that git blame tells me the line is from a rename, and then I would know to do git blame on the revision prior to that, if I really wanted to understand the purpose of a line.

Reply:
I was recently reading a committed change that also went and touched a ton of formatting in a large file, and it made reading the core of the actual change really hard, because the diff was massive, and it was harder to discern quickly the lines that actually made an impact for the true purpose of the change.

Reply:
If, however, the formatting changes were done in a separate commit, the change I was trying to understand would have been simpler, and anyone tracking that file could skim over the formatting changes commit, or skip it altogether, saving them time and energy.